### PR TITLE
FM2-408: Encounters should be sortable by date

### PR DESCRIPTION
**Pull request by Richa-09**
## Description of what I changed
Implemented sorting by date for Encounters

## Issue I worked on
see https://issues.openmrs.org/browse/FM2-408

## Checklist: I completed these to help reviewers :)
- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImpl.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
 @Component
 @Setter(AccessLevel.PACKAGE)
-public class FhirEncounterDaoImpl extends BaseFhirDao<Encounter> implements FhirEncounterDao {
+public class FhirEncounterDaoImpl extends BaseEncounterDao<Encounter> implements FhirEncounterDao {
```
Comment (jnsereko) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImpl.java:39]: Why change this?

**Review by jnsereko — COMMENTED** [review]
Nice work @Richa-09 
cc @ibacher

File: api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	
+	@Override
+	protected String paramToProp(@Nonnull String param) {
+		switch (param) {
+			case SP_DATE:
+				return "encounterDatetime";
+			default:
+				return null;
+		}
+	}
```
Comment (jnsereko) [inline api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java:35]: Why not override your method here; https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImpl.java

Comment (Richa-09) [reply api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java:35]: This seems better. I'll do the changes.

**Comment by Richa-09** [comment]
I have done the changes. PTAL @jnsereko @ibacher

**Review by jnsereko — APPROVED** [review]
LGTM 
I have to confirm from my senior though!
cc @ibacher

**Review by ibacher — APPROVED** [review]
@Richa-09 Thanks for this!

**Pull request merged**